zkevm-circuits/src/evm_circuit/op_execution/jumpi.rs

```diff
@@ -0,0 +1,351 @@
+};
+use super::utils::{
+    self,
+    common_cases::{OutOfGasCase, StackUnderflowCase},
+    constraint_builder::ConstraintBuilder,
+    math_gadgets::IsZeroGadget,
+    sum, StateTransition, StateTransitionExpressions,
+};
+
+use super::{CaseAllocation, CaseConfig, OpExecutionState, OpGadget};
+use crate::impl_op_gadget;
+use crate::util::{Expr, ToWord};
+use bus_mapping::evm::{GasCost, OpcodeId};
+use halo2::plonk::Error;
+use halo2::{arithmetic::FieldExt, circuit::Region};
+use num::BigUint;
+use num::ToPrimitive;
+use std::convert::TryInto;
+
+static STATE_TRANSITION: StateTransition = StateTransition {
+    gc_delta: Some(2),
+    pc_delta: Some(1), // pc is dynamic for jumpi, just set 1 for initialization
+    sp_delta: Some(2),
+    gas_delta: Some(GasCost::SLOW.as_u64()),
+    next_memory_size: None,
+};
+
+const NUM_POPPED: usize = 2;
+impl_op_gadget!(
+    #set[JUMPI]
+    JumpiGadget {
+        JumpiSuccessCase(),
+        OutOfGasCase(STATE_TRANSITION.gas_delta.unwrap()),
+        StackUnderflowCase(NUM_POPPED),
+        //TODO: ErrJumpcase
+    }
+);
+
+#[derive(Clone, Debug)]
+struct JumpiSuccessCase<F> {
+    case_selector: Cell<F>,
+    code_hash: Word<F>,
+    dest: Word<F>,
+    cond: Word<F>,
+    is_cond_zero: IsZeroGadget<F>,
+}
+
+impl<F: FieldExt> JumpiSuccessCase<F> {
+    pub(crate) const CASE_CONFIG: &'static CaseConfig = &CaseConfig {
+        case: Case::Success,
+        num_word: 3, // two stack values + code_hash of contract
+        num_cell: IsZeroGadget::<F>::NUM_CELLS,
+        will_halt: false,
+    };
+
+    pub(crate) fn construct(alloc: &mut CaseAllocation<F>) -> Self {
+        Self {
+            case_selector: alloc.selector.clone(),
+            code_hash: alloc.words.pop().unwrap(),
+            dest: alloc.words.pop().unwrap(),
+            cond: alloc.words.pop().unwrap(),
+            is_cond_zero: IsZeroGadget::construct(alloc),
+        }
+    }
+
+    pub(crate) fn constraint(
+        &self,
+        state_curr: &OpExecutionState<F>,
+        state_next: &OpExecutionState<F>,
+        name: &'static str,
+    ) -> Vec<Constraint<F>> {
+        let mut constrains = Vec::<Constraint<F>>::new();
+        let mut cb = ConstraintBuilder::default();
+
+        let is_cond_met = 1.expr()
+            - self
+                .is_cond_zero
+                .constraints(&mut cb, sum::expr(&self.cond.cells));
+
+        // State transitions
+        let st = StateTransitionExpressions::new(STATE_TRANSITION.clone());
+        // is_cond_met == 0 --> cond = 0, pc + 1
+        cb.require_zero(
+            (1.expr() - is_cond_met.clone())
+                * (st.clone().pc_delta.unwrap() - 1.expr()),
+        );
+
+        // is_cond_met == 1 --> cond != 0, pc = `dest`
+        cb.require_zero(
+            is_cond_met.clone()
+                * (st.clone().pc_delta.unwrap()
+                    - (self.dest.expr() - state_curr.program_counter.expr())),
+        );
```

Comment:
`st.pc_delta = select::expr(is_cond_met, self.dest.expr() - state_curr.program_counter.expr(), 1.expr())` makes more sense to me.

Reply:
yeah, make more sense & will update Thanks!

Reply:
updated in 3ef85a7